tests/ui/compiletest-self-test/test-aux-bin.rs

```diff
@@ -0,0 +1,8 @@
+//@ run-pass
+//@ aux-bin: print-it-works.rs
+
+fn main() {
+    let stdout =
+        std::process::Command::new("auxiliary/bin/print-it-works").output().unwrap().stdout;
```

Comment:
Not sure about the paths, but I think it might at the very least be missing .exe for Windows?

Worth editing PR CI to try this out.

Reply:
I have easy access to a windows machine and tried the test in windows before push, and yes, it works.

it works because in windows .exe is automatically  [checked](https://doc.rust-lang.org/std/process/struct.Command.html#platform-specific-behavior) for

Reply:
I just realized it will not work with cross compilation since we run the produced binary. I will add an ignore directive for that when I get home.

Reply:
Maybe aux-bin should require/imply no cross compiles?

Reply:
Also, can you also make sure you add a description of this directive in the dev-guide somewhere, just so there's a reference for the directive somewhere?

Reply:
> Maybe aux-bin should require/imply no cross compiles?

Hmm I think that could be taking it to far, since I can imagine ui tests that use aux-bin but does not actually run the produced binary. Maybe the test will just inspect the metadata of the produced binary, or something along those lines.

> Also, can you also make sure you add a description of this directive in the dev-guide somewhere, just so there's a reference for the directive somewhere?

Absolutely, see https://github.com/rust-lang/rustc-dev-guide/pull/1942